parquet-variant/src/main/java/org/apache/parquet/variant/UnknownVariantTypeException.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.parquet.variant;
+
+/**
+ * An exception indicating that the Variant contains an unknown type.
+ */
+public class UnknownVariantTypeException extends RuntimeException {
+  public final int typeId;
+
+  /**
+   * @param typeId the type id that was unknown
+   */
+  public UnknownVariantTypeException(int typeId) {
+    super("Unknown type in Variant. id: " + typeId);
+    this.typeId = typeId;
+  }
+
+  /**
+   * @return the type id that was unknown
+   */
+  public int getTypeId() {
```

Comment:
Nit: I don't recommend using `get` although it is popular. There is almost always a more specific verb that is better, and if not it can usually be omitted for a cleaner method name.